File: AtlasWeb/Core/Modules/Tibbi_Surec/Kadin_Dogum_Modulu/InfertilityFormViewModel.cs
//$113818DF
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class InfertilityServiceController
    {
        partial void PreScript_InfertilityForm(InfertilityFormViewModel viewModel, Infertility infertility, TTObjectContext objectContext)
        {
            Guid? activePatientObjectID = viewModel.GetSelectedPatientID();
            if (activePatientObjectID.HasValue)
            {
                Patient patient = objectContext.GetObject<Patient>(activePatientObjectID.Value);
                viewModel._Patient = patient;
            }

            ContextToViewModel(viewModel, objectContext);
        }
    }
}

namespace Core.Models
{
    public partial class InfertilityFormViewModel
    {
        public Patient _Patient
        {
            get;
            set;
        }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Kadin_Dogum_Modulu/InfertilityFormViewModel.ts
//$60044C33
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { Infertility } from "NebulaClient/Model/AtlasClientModel";
import { Patient } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class InfertilityFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.InfertilityFormViewModel, Core';

    @Type(() => Infertility)
    public _Infertility: Infertility = new Infertility();

    @Type(() => Patient)
    public Patients: Array<Patient> = new Array<Patient>();

    @Type(() => Patient)
    public _Patient: Patient = new Patient();
}
